datanode/networkhistory/service.go

```diff
@@ -68,10 +71,35 @@ func New(ctx context.Context, log *logging.Logger, chainID string, cfg Config, c
 		chainID:             chainID,
 		snapshotsCopyToPath: snapshotsCopyToPath,
 		datanodeGrpcAPIPort: datanodeGrpcAPIPort,
+		fw:                  fw,
 	}
 
 	if cfg.Publish {
 		var err error
+		go func() {
+
+			// consume all pending files
+			f := func() {
+				for !fw.Empty() {
+					if err = fw.Consume(); err != nil {
+						s.log.Error("failed to write all files to disk", logging.Error(err))
+					}
+				}
+			}
+
+			ticker := time.NewTicker(5 * time.Second)
+			for {
+				select {
+				case <-ctx.Done():
+					f()
+					return
+				case <-ticker.C:
+					f()
```

Comment:
why not call `PublishSegments` here, too (if there are any new segments to publish)?